Tests/MongoSwiftTests/RetryableWritesTests.swift

```diff
@@ -0,0 +1,109 @@
+import Foundation
+@testable import MongoSwift
+import Nimble
+import XCTest
+
+/// Struct representing a single test within a spec test JSON file.
+private struct RetryableWritesTest: Decodable, SpecTest {
+    let description: String
+    let outcome: TestOutcome
+    let operation: AnyTestOperation
+
+    /// Options used to configure the `MongoClient` used for this test.
+    let clientOptions: ClientOptions?
+
+    /// If true, the `MongoClient` for this test should be initialized with multiple mongos seed addresses.
+    /// If false or omitted, only a single mongos address should be specified.
+    /// This field has no effect for non-sharded topologies.
+    let useMultipleMongoses: Bool?
+
+    /// The optional fail point to configure before running this test.
+    /// This option and useMultipleMongoses: true are mutually exclusive.
+    let failPoint: FailPoint?
+}
+
+/// Struct representing a single retryable-writes spec test JSON file.
+private struct RetryableWritesTestFile: Decodable {
```

Comment:
The idea was to keep the framework minimal (at this point) so that crud v1 tests can also use it. I think I'd prefer to defer deciding on moving which parts into the framework once we implement v2 if that's okay. That work will probably involve modularizing the runner framework so that both v2 style and v1 style tests can use the parts they need without duplicating e.g. test file structs or unified test format based structs.

Reply:
SGTM